File: javaAndroid/UserAdapter.java
package bg.tcom.c12v1gr.demo122applivation;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.List;

public class UserAdapter extends ArrayAdapter<User> {
        private Context context;
        private int layoutId;
        private List<User> data;
        public UserAdapter(@NonNull Context context, int resource, @NonNull List<User> objects) {
            super(context, resource, objects);
            this.context = context;
            this.layoutId = resource;
            this.data = objects;
        }

        @NonNull
        @Override
        public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
            LayoutInflater layoutInflater = ((Activity)context).getLayoutInflater();
            View row = layoutInflater.inflate(layoutId, parent, false);

            TextView tw1 = row.findViewById(R.id.twName);
            tw1.setText(data.get(position).getUserName());

            TextView tw2 = row.findViewById(R.id.twPop);
            tw2.setText(data.get(position).geteMail());

            return row;
        }
    }

File: csAndroid/2024-11-14/UserAdapter.cs
using Android.App;
using Android.Views;
using Android.Widget;
using System.Collections.Generic;
using Android.Content;

namespace App2
{
    public class User
    {
        public string UserName { get; set; }
        public string Mail { get; set; }
    }

    public class UserAdapter : ArrayAdapter<User>
    {
        private Context context;
        private int layoutId;
        private List<User> data;
        public UserAdapter(Context context, int resource, List<User> objects) : base(context, resource, objects)
        {

            this.context = context;
            this.layoutId = resource;
            this.data = objects;
        }

        public override View GetView(int position, View convertView, ViewGroup parent)
        {
            LayoutInflater layoutInflater = ((Activity)context).LayoutInflater;
            View row = layoutInflater.Inflate(layoutId, parent, false);

            TextView tw1 = row.FindViewById<TextView>(Resource.Id.twName);
            tw1.Text = data[position].UserName;

            TextView tw2 = row.FindViewById<TextView>(Resource.Id.twPop);
            tw2.Text = data[position].Mail;

            return row;
        }
    }

}
